### PR TITLE
Fix continuous batching in `transformers serve`

### PR DESCRIPTION
**Pull request by LysandreJik**

### CODE REVIEW

File: src/transformers/commands/serving.py [hunk 1]
```diff
@@ -326,54 +328,61 @@ def get_all_models():
...
+                max_batch_tokens=10,
+                scheduler="fifo",
             )
 
-            generation_config = create_generation_config_from_req(req)
+            if self.running_continuous_batching_manager is None or update_model:
+                self.running_continuous_batching_manager = self.model.init_continuous_batching(
+                    generation_config=generation_config, streaming=True
+                )
+                self.running_continuous_batching_manager.logit_processor = LogitsProcessorList()
```
Comment (LysandreJik) [inline src/transformers/commands/serving.py:400]: This is not great but the Qwen models currently fail because of the logits processors they use

Comment (gante) [reply src/transformers/commands/serving.py:400]: Shall we add a TODO to remove these lines then?

File: src/transformers/commands/serving.py [hunk 2]
```diff
@@ -326,54 +328,61 @@ def get_all_models():
...
+                self.running_continuous_batching_manager.start()
+
+            inputs = self.tokenizer.apply_chat_template(
```
Comment (gante) [inline src/transformers/commands/serving.py:404]: Suggestion: add a TODO to add tool support (in a nutshell, abstract and share the logic with the base `generate`)

File: src/transformers/commands/serving.py [hunk 3]
```diff
@@ -326,54 +328,61 @@ def get_all_models():
...
                                 continue
                             else:
                                 queue_is_flushed = True
 
                         finish_reason = "stop" if result.status == RequestStatus.FINISHED else None
-                        yield self.build_chunk(result.next_token, request_id=request_id, finish_reason=finish_reason)
+                        next_token = self.build_chunk(
+                            result.next_token, request_id=request_id, finish_reason=finish_reason
+                        )
+                        yield next_token
```
Comment (gante) [inline src/transformers/commands/serving.py]:
```suggestion
                        next_chunk = self.build_chunk(
                            result.next_token, request_id=request_id, finish_reason=finish_reason
                        )
                        yield next_chunk
```
(`next_token` is a chunk)

**Review by gante — APPROVED** [review]
LGTM, thank you for fixing 👍

File: src/transformers/commands/serving.py (hunk 3 above)
Comment (LysandreJik) [reply src/transformers/commands/serving.py]: Indeed!
